frontend/src/metabase/query_builder/components/QueryModals.tsx

```diff
@@ -0,0 +1,356 @@
+import React from "react";
+import { connect } from "react-redux";
+
+import { t } from "ttag";
+import _ from "underscore";
+
+import Questions from "metabase/entities/questions";
+import { ROOT_COLLECTION } from "metabase/entities/collections";
+
+import { MODAL_TYPES } from "metabase/query_builder/constants";
+
+import Modal from "metabase/components/Modal";
+
+import SaveQuestionModal from "metabase/containers/SaveQuestionModal";
+import QuestionSavedModal from "metabase/components/QuestionSavedModal";
+import AddToDashSelectDashModal from "metabase/containers/AddToDashSelectDashModal";
+
+import CollectionMoveModal from "metabase/containers/CollectionMoveModal";
+import ArchiveQuestionModal from "metabase/questions/containers/ArchiveQuestionModal";
+import QuestionEmbedWidget from "metabase/query_builder/containers/QuestionEmbedWidget";
+
+import { CreateAlertModalContent } from "metabase/query_builder/components/AlertModals";
+import { ImpossibleToCreateModelModal } from "metabase/query_builder/components/ImpossibleToCreateModelModal";
+import NewDatasetModal from "metabase/query_builder/components/NewDatasetModal";
+import EntityCopyModal from "metabase/entities/containers/EntityCopyModal";
+import BulkFilterModal from "metabase/query_builder/components/filters/modals/BulkFilterModal";
+import NewEventModal from "metabase/timelines/questions/containers/NewEventModal";
+import EditEventModal from "metabase/timelines/questions/containers/EditEventModal";
+import MoveEventModal from "metabase/timelines/questions/containers/MoveEventModal";
+import PreviewQueryModal from "metabase/query_builder/components/view/PreviewQueryModal";
+import ConvertQueryModal from "metabase/query_builder/components/view/ConvertQueryModal";
+import QuestionMoveToast from "metabase/questions/components/QuestionMoveToast";
+import { Card, Collection, User } from "metabase-types/api";
+import StructuredQuery from "metabase-lib/queries/StructuredQuery";
+import Question from "metabase-lib/Question";
+
+const mapDispatchToProps = {
+  setQuestionCollection: Questions.actions.setCollection,
+};
+
+type ModalType = typeof MODAL_TYPES[keyof typeof MODAL_TYPES];
+
+interface QueryModalsProps {
+  questionAlerts: any;
```

Comment:
Can we use a real type here?

Reply:
can you please help me with types here? I didn't find actual places from where to grab the type

Reply:
Yeah, seems like we don't have types for alerts yet. #30457 should add them; feel free to cherry-pick the commits here or we can wait until its approved and merged